Classic_RPG/src/entities/character/Ninja.java

```diff
@@ -3,54 +3,61 @@
 import entities.Entity;
 import entities.Player;
 import game_management.Interfaces.GUI;
+import game_management.random.Random;
+import game_management.random.TestOutput;
 import game_management.Action_Enums.Attack;
 import game_management.Action_Enums.Game_action;
 
-public class Ninja extends Player{
+public class Ninja extends Player {
 	final static int BASE_HP = 1000;
 	final static int BASE_STAMINA = 3000;
-	
+
 	final static int BASE_STRENGTH = 10;
 	final static int BASE_AGILITY = 20;
 	final static int BASE_INTELLIGENCE = 10;
-	
+
 	public Ninja(String name) {
 		super(BASE_HP, BASE_STAMINA, name, BASE_AGILITY, BASE_STRENGTH, BASE_INTELLIGENCE);
 	}
-	
+
 	/* Should not be pared */
 	public void specialHit(Entity target) {
+		int dmgs =  this.getCharacteristics().getStrength() * 4 * this.getCharacteristics().getAgility() / 5;
+		int success = getCharacteristics().getAgility() + 75;
+		int specialDamage = Random.EffectiveDamage(success, dmgs);
 		GUI.edit_message(name + " use surgical attack on " + target.getName());
-		attack(target, this.getCharacteristics().getStrength() * 4 * this.getCharacteristics().getAgility() / 5);
+		attack(target, specialDamage);
 		staminaChange(false, 300);
 	}
-	
+
 	public void basicHit(Entity target) {
 		int dmgs = getCharacteristics().getStrength() * 4 * this.getCharacteristics().getAgility() / 10;
-		
+		int success = getCharacteristics().getAgility() + 75;
+		Random.EffectiveDamage(success, dmgs);
 		GUI.edit_message(name + " basic attack on " + target.getName());
 		staminaChange(false, 130);
 		this.attack(target, dmgs);
 	}
-	
+
 	public void receiveAttack(int dmgIncoming) {
-		if(defense_position && getStamina() >= 150) {
+		int success = getCharacteristics().getAgility() + 75;
+		TestOutput result1D100 = Random.test1D100(success);
+		if (defense_position && getStamina() >= 150 && result1D100 != TestOutput.CRITICAL_FAILURE) {
 			staminaChange(false, 150);
 			GUI.edit_message(this.name + " did a beautiful dodge roll");
-		}
-		else {
+		} else {
```

Comment:
Add a case if TestOutput is FAILLURE, maybe take dmgs/2
Maybe something to do with a CRITICAL_SUCCESS ? free roll ? earn stamina ?